src/commands/cna/create-auth.js

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const CNABaseCommand = require('../../CNABaseCommand')
+const { cli } = require('cli-ux')
+const path = require('path')
+const fs = require('fs-extra')
+const spawn = require('cross-spawn')
+const which = require('which')
+const config = require('@adobe/aio-cli-config')
+const inquirer = require('inquirer')
+
+class CNACreateAuth extends CNABaseCommand {
+  async run () {
+    this.parse(CNACreateAuth)
+
+    // 1. make path absolute
+    let destDir = path.resolve('./adobeio-cna-lib-auth-ims/')
+
+    // 2. Make sure we have npm, fatal otherwise
+    if (!_isNpmInstalled()) { // todo: better error message
+      this.error('npm is required.')
+    }
+
+    // 3. Make sure we have git, fatal otherwise
+    if (!_isGitInstalled()) { // todo: better error message
+      this.error('git is required.')
+    }
+
+    // 4. fail if destination dir exists
+    if (fs.existsSync(destDir)) {
+      this.warn('A folder with name adobeio-cna-lib-auth-ims already exists. Skipping download.')
+    } else {
+      await cloneRepo()
+      await installDeps(destDir)
+    }
+
+    // 5. Just do it
+    let IMS_AUTH_TYPE = await _getUserInput({
+      name: 'IMS_AUTH_TYPE',
+      message: 'Which mode of auth do you need? (Valid options are code and jwt',
+      type: 'string',
+      default: 'jwt'
+    })
+    let IMS_AUTH_PERSIST = await _getUserInput({
+      name: 'IMS_AUTH_PERSIST',
+      message: 'Do you want to persist the tokens in db? (yes or no)',
+      type: 'string',
+      default: 'no'
+    })
+    await configureAuth(destDir, IMS_AUTH_TYPE, IMS_AUTH_PERSIST)
+    await IMS_AUTH_TYPE === 'code' ? deployAuth(destDir) : deployJWTAuth(destDir)
+  }
+}
+
+async function cloneRepo () {
+  return new Promise(function (resolve, reject) {
+    cli.action.start('cloning auth library')
+    const child = spawn('git', ['clone', 'https://github.com/adobe/adobeio-cna-lib-auth-ims'])
```

Comment:
This means we could be creating a git repo inside the developers app, which might also be in a git repo.
I think ultimately we just want the action source files and some additional config added to the developers project, and they could then commit that to their own version control.